### DISCUSSION TITLE
Perplexica cannot find ollama server on Ubuntu 24.04 using docker

### DISCUSSION THREAD

**Discussion by ahmetax** [ItzCrazyKns/Vane#402] [General]
Hi,
I installed Perplexica using docker on my Ubuntu 24.04 system. Ollama and seaxng is already running.
However "docker compose up -d " does not run on my system. Instead, I run "docker-compose up -d".
When I call "http://localhost:3000" on my browser it says "Failed to connect to the server. Please try again later."
What can I do?
![Screenshot from 2024-10-11 10-01-00](https://github.com/user-attachments/assets/1171b61d-76b7-4e41-be2b-a250ca5e2260)

**Comment by ItzCrazyKns** [comment]
Can you show me what Ollama address are you using in the config, there must be some issues with it since its not able to request the models from it.

**Reply by ahmetax** [reply]
I tried both http://localhost:11434 and http://host.docker.internal:11434. Same results for both.  Loaded models list is in the attached image:
![Screenshot from 2024-10-12 03-50-11](https://github.com/user-attachments/assets/a6af4c98-ede7-46fe-b8f0-030ec151ee33)

**Reply by ItzCrazyKns** [reply]
Can you try going to localhost:11434 and see what pops up?

**Reply by ahmetax** [reply]
It says "Ollama is running"
I also upgraded docker to the latest version. Removed all existing images and volumes. Then applied "docker compose up -d" command.
But, nothing changed!
searxng is running at http://localhost:4000 (not at port 32768)
http://localhost:3000 is open and running and saying "Failed to connect to the server..." When pressed "Settings" button, a continuously rotating shape is seen without any progress.

**Reply by ahmetax** [reply]
At last I found my problem!
I am not a docker fan. I only followed the given instructions.
After trying many re-installations, I figured out when I removed config.yaml, behaviour of Perplexica didn't change. Then I searched for a command to restart docker containers.
And, I found the command: docker restart $(docker ps -a -q)
Bingo!
Now, Perplexica runs!
But, now, only Chat model Provider is Custom_openai. And it cannot make any searchs.
How can I activete searxng and ollama?
I couldn't find a solution.

**Reply by ItzCrazyKns** [reply]
You need to follow [this](https://github.com/ItzCrazyKns/Perplexica?tab=readme-ov-file#ollama-connection-errors) guide
